### PR TITLE
[interp] Fix memory overwrite during pinvoke r4 parameter passing.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/19684.

### CODE REVIEW

**Comment by vargaz** [comment]
This only affects wasm, other platforms use a different implementation.
@lewing this might need a backport.

**Comment by lewing** [comment]
@monojenkins backport 3.2-wasm

**Comment by vargaz** [comment]
The test failures are related, there is a problem with the patch.

**Comment by vargaz** [comment]
This should work now. The previous code was using 2 slots for doubles for some reason, maybe fargs was a float array previously ?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by radical** [comment]
@monojenkins build failed

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -1384,11 +1384,7 @@ static InterpMethodArguments* build_args_from_sig (MonoMethodSignature *sig, Int
 #if DEBUG_INTERP
 			g_print ("build_args_from_sig: margs->fargs [%d]: %p (%f) (frame @ %d)\n", int_f, margs->fargs [int_f], margs->fargs [int_f], i);
 #endif
-#if SIZEOF_VOID_P == 4
-			int_f += 2;
-#else
-			int_f++;
-#endif
+			int_f ++;
```
Comment (BrzVlad) [inline mono/mini/interp/interp.c:1387]: We are still allocating 2 slots for R8 in the initial loop. Is this just hiding the real issue ?

Comment (vargaz) [reply mono/mini/interp/interp.c:1387]: For r4, we only allocate 1 slot.

Comment (BrzVlad) [reply mono/mini/interp/interp.c:1387]: Shouldn't we always allocate only 1 slot for both r4 and r8 (flen from above)

Comment (vargaz) [reply mono/mini/interp/interp.c:1387]: Yes

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**